redis/cluster.py

```diff
@@ -1561,25 +1561,25 @@ def initialize(self):
                 # add this node to the nodes cache
                 tmp_nodes_cache[target_node.name] = target_node
 
+                target_replica_nodes = []
+                for replica_node in slot[3:]:
+                    host = str_if_bytes(replica_node[0])
+                    port = int(replica_node[1])
+                    host, port = self.remap_host_port(host, port)
+
+                    target_replica_node = self._get_or_create_cluster_node(
+                        host, port, REPLICA, tmp_nodes_cache
+                    )
+                    target_replica_nodes.append(target_replica_node)
+                    # add this node to the nodes cache
+                    tmp_nodes_cache[target_replica_node.name] = target_replica_node
+
                 for i in range(int(slot[0]), int(slot[1]) + 1):
                     if i not in tmp_slots:
                         tmp_slots[i] = []
                         tmp_slots[i].append(target_node)
-                        replica_nodes = [slot[j] for j in range(3, len(slot))]
```

Comment:
```py
replica_nodes = slot[3:]
```

if we change this line like above, how much does performance change?

Reply:
because all below lines should not affect performance a lot, I believe. All of its time complexity is O(1)

Reply:
`replica_nodes = slot[3:]` improves the time spent up to `11.043858208024176`

the bigger difference came from not iterating through replica nodes by the number of slots

Reply:
i know your change is more neat, just curious for python performance

Reply:
> the bigger difference came from not iterating through replica nodes by the number of slots

Hmm? but isn't replica nodes traversed by the number of slots still in your change?

Reply:
that part is true, but looking for node doesn't have to be done in those iterations
```
host = str_if_bytes(replica_node[0])
port = replica_node[1]
host, port = self.remap_host_port(host, port)

target_replica_node = self._get_or_create_cluster_node(host, port, REPLICA, tmp_nodes_cache)
```

Reply:
🤔 🤔 🤔 What's wrong with you python. why the 16384 * above operations bring that much performance difference 🐍